aws_helper/config.go

```diff
@@ -17,7 +17,7 @@ import (
 func CreateAwsSession(awsRegion, awsEndpoint string, awsProfile string, iamRoleArn string, terragruntOptions *options.TerragruntOptions) (*session.Session, error) {
 	defaultResolver := endpoints.DefaultResolver()
 	s3CustResolverFn := func(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
-		if service == "s3" {
+		if service == "s3" && awsEndpoint != "" {
```

Comment:
Ah, interesting. Should the param be called `customS3Endpoint` or similar to make it clearer?

Reply:
Yes, I think this is a good idea!
But only in `func CreateAwsSession`?

Reply:
Probably in `CreateS3Client` too.

Reply:
Ok, why not, so it's same name overall the code!
I'll do the changes right away.